central/graphql/resolvers/node_components.go

```diff
@@ -0,0 +1,172 @@
+package resolvers
+
+import (
+	"context"
+	"time"
+
+	"github.com/graph-gophers/graphql-go"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/metrics"
+	pkgMetrics "github.com/stackrox/rox/pkg/metrics"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/utils"
+)
+
+func init() {
+	schema := getBuilder()
+	utils.Must(
+		// NOTE: This list should be alphabetically ordered
+		schema.AddType("NodeComponent", []string{
+			"fixedIn: String!", // is this needed?
+			"id: ID!",
+			"lastScanned: Time",
+			"license: License",
+			"location(query: String): String!",
+			"name: String!",
+			"nodes(query: String, scopeQuery: String, pagination: Pagination): [Node!]!",
+			"nodeCount(query: String, scopeQuery: String): Int!",
+			"plottedVulnerabilities(query: String): PlottedVulnerabilities!",
+			"priority: Int!",
+			"riskScore: Float!",
+			"source: String!", // is this infrastructure ?
+			"topVulnerability: NodeVulnerability",
+			"unusedVarSink(query: String): Int",
+			"version: String!",
+			"vulnerabilities(query: String, scopeQuery: String, pagination: Pagination): [NodeVulnerability]!",
+			"vulnerabilityCount(query: String, scopeQuery: String): Int!",
+			"vulnerabilityCounter(query: String): VulnerabilityCounter!",
+		}),
+		// TODO : Tie this to NodeScan resolvers
+		// schema.AddExtraResolver("NodeScan", `nodeComponents(query: String, pagination: Pagination): [NodeComponent!]!`),
+		// schema.AddExtraResolver("NodeScan", `componentCount(query: String): Int!`),
+		schema.AddQuery("nodeComponent(id: ID): NodeComponent"),
+		schema.AddQuery("nodeComponents(query: String, scopeQuery: String, pagination: Pagination): [NodeComponent!]!"),
+		schema.AddQuery("nodeComponentCount(query: String): Int!"),
```

Comment:
Looks like you didn't implement this function?

Reply:
Yeah I forgot to add that. Just added it.